src/main/java/net/starschema/clouddb/jdbc/BQSupportFuncts.java

```diff
@@ -364,14 +365,19 @@ public static String logAndGetQueryState(Job pollJob) throws IOException {
         || pollJob.getStatistics().isEmpty()) {
       throw new IOException("Failed to fetch query state.");
     }
-    BQSupportFuncts.logger.info(
-        "Job status: "
-            + pollJob.getStatus().getState()
-            + " ; "
-            + pollJob.getJobReference().getJobId()
-            + " ; "
-            + (System.currentTimeMillis() - pollJob.getStatistics().getStartTime()));
-    return pollJob.getStatus().getState();
+
+    if (Optional.ofNullable(pollJob.getStatistics().getCreationTime()).isPresent()) {
+      BQSupportFuncts.logger.info(
+          "Job status: "
+              + pollJob.getStatus().getState()
+              + " ; "
+              + pollJob.getJobReference().getJobId()
+              + " ; "
+              + (System.currentTimeMillis() - pollJob.getStatistics().getCreationTime()));
```

Comment:
do we actually want to log creation time or start time?

Reply:
(which is preferred and why -- please comment)
Is creationTime always set but startTime isn't?

Reply:
According to the internal bug: when a query is queued, its statistics do not include 'startTime', but they do include 'creationTime'. So when the logger previously tried to calculate the elapsed time, an NPE was thrown (hence the switch).

Reply:
I think this is trying to log the elapsed time of the job, so falling back to creation_time in the absence of start_time makes sense.  As mentioned in another comment, the creation_time should not be empty.